src/main/java/org/craftercms/studio/impl/v2/dal/StudioDBScriptRunnerImpl.java

```diff
@@ -61,16 +64,40 @@ public void closeConnection() {
     }
 
     @Override
-    public void execute(String sql) {
-        try (Reader reader = new StringReader(sql)) {
-            Connection connection = dataSource.getConnection();
+    public void execute(File file) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            connection = dataSource.getConnection();
+
+            connection.setAutoCommit(false);
+
             ScriptRunner scriptRunner = new ScriptRunner(connection);
+            scriptRunner.setAutoCommit(false);
             scriptRunner.setDelimiter(delimiter);
             scriptRunner.setStopOnError(true);
             scriptRunner.setLogWriter(null);
-            scriptRunner.runScript(reader);
+
+            StringBuilder sb = new StringBuilder();
```

Comment:
StringBuilder sb = new StringBuilder();
String line = null;
Boolean moreWork = true;

while (moreWork) {
	for (i = 0; i < scriptLinesBufferSize && moreWork; i++) {
		line = bufferedReader.readLine;
		if (line != null) {
			sb.append(line).append("\n");
		} else {
			moreWork = false;
		}	
	}
	
	if (sb.length() > 0) {
		scriptRunner.runScript(new StringReader(sb.toString()));
		sb.setLength(0);
	}
}